syntaxes/systemverilog.tmLanguage.yaml

```diff
@@ -80,33 +70,34 @@ repository:
         name: keyword.control.systemverilog
       '4':
         name: storage.type.systemverilog
-    end: '(?<=}\s*)([a-zA-Z_][a-zA-Z0-9_$]*)(?:\s*(\[[a-zA-Z0-9_:\$\-\+\s*]*\])?)\s*;'
+    end: (?<=})(?:[ \t\n]*([a-zA-Z_][a-zA-Z0-9_$]*))(?:[ \t\n]*(\[[a-zA-Z0-9_:$\-\+` \t\n\[\]]*\])?)[ \t\n]*;
     endCaptures:
       '1':
         name: storage.type.systemverilog
       '2':
         patterns:
-          - include: '#slices'
+          - include: '#selects'
     patterns:
-      # - include: '#struct-anonymous'
+      - include: '#keywords'
       - include: '#base-grammar'
+      - include: '#identifiers'
     name: meta.typedef.complex.systemverilog
   typedef-simple:
-    begin: \s*(typedef)\s+(enum|struct|union|class|interface\s+class)\b
+    begin: '[ \t\n]*\b(typedef)[ \t\n]+(enum|struct|union|class|interface[ \t\n]+class)\b'
     beginCaptures:
       '1':
         name: keyword.control.systemverilog
       '2':
         name: keyword.control.systemverilog
-    end: '([a-zA-Z_][a-zA-Z0-9_$]*)(?:\s*(\[[a-zA-Z0-9_:\$\-\+\s*]*\])?)\s*;'
+    end: (?:[ \t\n]*([a-zA-Z_][a-zA-Z0-9_$]*))(?:[ \t\n]*(\[[a-zA-Z0-9_:$\-\+` \t\n\[\]]*\])?)[ \t\n]*;
     endCaptures:
       '1':
         name: storage.type.systemverilog
       '2':
         patterns:
-          - include: '#slices'
+          - include: '#selects'
     patterns:
-      - match: '\b([a-zA-Z_][a-zA-Z0-9_$]*)\s*(#)\('
+      - match: '[ \t\n]+\b([a-zA-Z_][a-zA-Z0-9_$]*)[ \t\n]*(#)\('
```

Comment:
just noting that CR is not supported, I think that is fine. LF should be the enforced standard for any serious coder anyway.

Reply:
From section 5.3 of the LRM:

> White space shall contain the characters for spaces, tabs, newlines, and formfeeds.

FF is distinct from CR and I don't think relevant to VS Code which renders a small `FF` placeholder if `editor.renderControlCharacters` is enabled. But really the interpretation for the regular expression comes from Annex A A.9.4 where it is defined as:

> white_space ::= space | tab | newline | eof

I interpret newline to mean the common name for LF and not CR LF. The only argument for using CR is compatibility. I would still prefer to keep the character class instead of `\s` because FF and VT are of no practical use. What do you think?

Reply:
I think using `[ \t\n]` is alright.

Would be nice if  `\s` would include newline as default, or perhaps a `\S` that included newline

Reply:
For future reference, in Oniguruma Regular Expressions (the variant used in VS Code for syntax highlighting), `\s` is defined as:

```text
  \s       whitespace char

           Not Unicode:
             \t, \n, \v, \f, \r, \x20

           Unicode case:
             U+0009, U+000A, U+000B, U+000C, U+000D, U+0085(NEL),
             General_Category -- Line_Separator
                              -- Paragraph_Separator
                              -- Space_Separator
```